elfo-configurer/src/lib.rs

```diff
@@ -76,6 +76,34 @@ impl Configurer {
     }
 
     async fn main(mut self) {
+        let mut first_envelope = match self.ctx.try_recv().await {
+            Ok(e) => {
+                msg!(match e {
+                    // We do not expect the first message to be `ConfigUpdated` because when the
+                    // actor is just started, `UpdateConfig` is consumed by the supervisor.
+                    (StartEntrypoint { is_check_only, .. }, token) => {
+                        if is_check_only {
+                            // We deserialize configs without sending any `ValidateConfig` messages
+                            // to avoid spawning any new actors.
+                            match self.load_configs().await {
+                                Ok(_) => self.ctx.respond(token, Ok(())),
+                                Err(e) => self
+                                    .ctx
+                                    .respond(token, Err(EntrypointFailed::new(e.reason))),
+                            }
+                            return;
+                        }
+                        None
+                    }
+                    // In case the actor was restarted by the supervisor, no `StartEntrypoint`
+                    // message is sent and we need to process any incoming envelopes normally.
+                    e => Some(e),
+                })
+            }
+            Err(TryRecvError::Empty) => None,
+            Err(TryRecvError::Closed) => panic!("failed to recv first envelope"),
```

Comment:
This group doesn't override the termination policy, so on `Terminate` just after restarting, it will get `Closed` here and panic. I prefer `return;` here